tests/foreman/api/test_syncplan.py

```diff
@@ -371,21 +370,25 @@ def test_positive_update_interval(self):
         :CaseImportance: Critical
         """
         for interval in valid_sync_interval():
-            with self.subTest(interval):
-                sync_plan = entities.SyncPlan(organization=self.org)
-                result = sync_plan.get_fields()['interval']
-                sync_plan.interval = sample(
-                    set(result.choices) - set([interval]),
-                    1
-                )[0]
-                sync_plan = sync_plan.create()
-                sync_plan.interval = interval
-                if (interval == 'custom cron'):
-                    sync_plan.cron_expression = gen_choice(valid_cron_expressions())
-                    sync_plan = sync_plan.update(['interval', 'cron_expression'])
-                else:
-                    sync_plan = sync_plan.update(['interval'])
-                self.assertEqual(sync_plan.interval, interval)
+            sync_plan = entities.SyncPlan(
+                description=gen_string('alpha'),
+                organization=self.org,
+                interval=interval
+            )
+            if interval == SYNC_INTERVAL['custom']:
+                sync_plan.cron_expression = gen_choice(valid_cron_expressions())
+            sync_plan = sync_plan.create()
+
+            valid_intervals = valid_sync_interval()
+            valid_intervals.remove(interval)
+            new_interval = valid_intervals[0]
+            sync_plan.interval = new_interval
+            if new_interval == SYNC_INTERVAL['custom']:
+                sync_plan.cron_expression = gen_choice(valid_cron_expressions())
+                sync_plan = sync_plan.update(['interval', 'cron_expression'])
+            else:
+                sync_plan = sync_plan.update(['interval'])
```

Comment:
Overall it looks good to me, if you share test result I'm ok with merging. iiuc there could be three update situations in this test:

non-custom > non-custom
non-custom > custom
custom > non-custom

I'm a bit concerned about the third one, If I create custom syncplan with cron expression and then change the interval but won't touch the cron experssion, will that succeed? If no, we'll need to clean cron_expression in that else statement. Anyway, test result should show

Reply:
Hello @pondrejk , thank you for your review. Tests pass locally but Travis build failed. I am going to change the section to use gen_choice, but here is test result before I do that:

 robottelo]$ pytest -pdb tests/foreman/api/test_syncplan.py::SyncPlanUpdateTestCase::test_positive_update_interval
===================== test session starts =================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.6.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo, inifile:
plugins: xdist-1.23.2, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting ... 2019-02-22 10:31:07 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                             

tests/foreman/api/test_syncplan.py .                                                                                   [100%]

======================== 1 passed in 5.05 seconds ==========================